pkg/blobstore/readfallback/read_fallback_blob_access.go

```diff
@@ -90,5 +91,23 @@ func (ba *readFallbackBlobAccess) FindMissing(ctx context.Context, digests diges
 	if err != nil {
 		return digest.EmptySet, util.StatusWrap(err, "Secondary")
 	}
+
+	// Replicate the blobs that are present only in the secondary backend to primary
+	if ba.replicator != nil {
```

Comment:
This if-statement can be removed. The Buildbarn codebase tries to follow the [null object pattern](https://en.wikipedia.org/wiki/Null_object_pattern) where possible. So if you specify a 'noop' blob replicator, it's actually an instance of this type:

https://github.com/buildbarn/bb-storage/blob/master/pkg/blobstore/replication/noop_blob_replicator.go

Reply:
I've added this check because of the description of a previous pull request touching this file, mentioning backwards compatibility (https://github.com/buildbarn/bb-storage/commit/47ea975ed072816cb5a2c74f9fe563113d1a9a76) and because there is a nil check in other places in this file.
Is this still safe to remove with this info?

Reply:
It's still safe to remove it. What's in the commit message likely applied to the initial version of the PR. See this comment in the PR, where I requested the same thing:

https://github.com/buildbarn/bb-storage/pull/79#discussion_r468105709

Reply:
Thank you for clarifying! Let me update!